vcd/datasource_vcd_network_test.go

```diff
@@ -0,0 +1,367 @@
+					resource.TestCheckOutput("network_edge", data.parent),
+					resource.TestCheckOutput("default_gateway", testConfig.Networking.ExternalNetwork),
+				),
+			},
+		},
+	})
+}
+
+func TestAccVcdNetworkIsolatedDS(t *testing.T) {
+	// This test requires access to the vCD before filling templates
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	err := getAvailableNetworks()
+
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		t.Skip("error getting available networks")
+		return
+	}
+	if len(availableNetworks) == 0 {
+		t.Skip("No networks found - data source test skipped")
+		return
+	}
+
+	networkType := "vcd_network_isolated"
+	data, ok := availableNetworks[networkType]
+	if !ok {
+		t.Skip("no isolated network found ")
+		return
+	}
+
+	template := networkTemplates[networkType]
+	var params = StringMap{
+		"Org":         testConfig.VCD.Org,
+		"VDC":         testConfig.VCD.Vdc,
+		"NetworkName": data.network.Name,
+		"FuncName":    "TestNetworkIsolatedDS",
+		"Tags":        "network",
+	}
+	configText := templateFill(template, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { preRunChecks(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckOutput("network_org", testConfig.VCD.Org),
+					resource.TestCheckOutput("network_vdc", testConfig.VCD.Vdc),
+					resource.TestCheckOutput("network_name", data.network.Name),
+					resource.TestCheckOutput("network_gateway", data.network.Configuration.IPScopes.IPScope[0].Gateway),
+					resource.TestCheckOutput("network_netmask", data.network.Configuration.IPScopes.IPScope[0].Netmask),
+					resource.TestCheckOutput("network_start_address", data.network.Configuration.IPScopes.IPScope[0].IPRanges.IPRange[0].StartAddress),
+				),
+			},
+		},
+	})
+}
+
+const datasourceTestNetworkDirect = `
+data "vcd_network_direct" "{{.NetworkName}}" {
+  name             = "{{.NetworkName}}"
+  org              = "{{.Org}}"
+  vdc              = "{{.VDC}}"
+  external_network = "{{.ExternalNetwork}}"
+}
+
+output "network_name" {
+  value = data.vcd_network_direct.{{.NetworkName}}.name
+}
+
+output "network_org" {
+  value = data.vcd_network_direct.{{.NetworkName}}.org
+}
+
+output "network_vdc" {
+  value = data.vcd_network_direct.{{.NetworkName}}.vdc
+}
+
+output "external_network" {
+  value = data.vcd_network_direct.{{.NetworkName}}.external_network
+}
+`
+
+const datasourceTestNetworkRouted = `
+data "vcd_edgegateway" "{{.EdgeGateway}}" {
+  name = "{{.EdgeGateway}}"
+  org  = "{{.Org}}"
+  vdc  = "{{.VDC}}"
+}
+
+data "vcd_network_routed" "{{.NetworkName}}" {
+  name         = "{{.NetworkName}}"
+  org          = "{{.Org}}"
+  vdc          = "{{.VDC}}"
+  edge_gateway = "${data.vcd_edgegateway.{{.EdgeGateway}}.name}"
```

Comment:
Per our discussions `edge_gateway` shouldn't be needed for routed network lookup, as we should be able to find it automatically, no?

Reply:
removed